### PR TITLE
Documentation page for gRPC

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: docs/wire_grpc.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+implementation("com.squareup.wire:wire-runtime:LATEST_VERSION")
+```
+
+In addition, to use Wire gRPC for clients, use these new Gradle coordinates:
+
+```groovy
+implementation("com.squareup.wire:wire-grpc-client:LATEST_VERSION")
+```    
+
+ [gradlePlugin]: https://square.github.io/wire/wire_compiler/
```
Comment (oldergod) [inline docs/wire_grpc.md]: Can't I just shortcut all those `https://square.github.io/wire/` with relative path?

File: docs/wire_grpc.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+ [grpcCall]: https://square.github.io/wire/3.x/wire-grpc-client/com.squareup.wire/-grpc-call/
+ [grpcClient]: https://square.github.io/wire/3.x/wire-grpc-client/com.squareup.wire/-grpc-client/
+ [grpcStreamingCall]: https://square.github.io/wire/3.x/wire-grpc-client/com.squareup.wire/-grpc-streaming-call/
+ [messageSink]: https://square.github.io/wire/3.x/wire-runtime/com.squareup.wire/-message-sink/
+ [messageSource]: https://square.github.io/wire/3.x/wire-runtime/com.squareup.wire/-message-source/
+ [sampleProject]: https://github.com/square/wire/tree/master/wire-grpc-sample
```
Comment (oldergod) [inline docs/wire_grpc.md]: Can't I just shortcut all those https://square.github.io/wire/ with relative path?

File: docs/wire_grpc.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+Wire gRPC
+=============================
+
+The Wire compiler will generate interfaces for your service RPCs defined in your protobuf schema;
+under the condition that the target is set to Kotlin.
+
+Getting Started
+---------------
+
+You can configure how your services are generated via our [Gradle plugin][gradlePlugin]:
```
Comment (Egorand) [inline docs/wire_grpc.md]:
```suggestion
You can configure how your services are generated via the [Gradle plugin][gradlePlugin]:
```

File: docs/wire_grpc.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  kotlin { // Kotlin target will generate code for services.
+    // `client` to generate interfaces best suited to sending outbound calls.
+    // `server` to generate interfaces best suited to receiving inbound calls.
+    rpcRole = 'server'
+    // Server only
+    // `suspending` to generate coroutines APIs that require a Kotlin coroutines context.
+    // `blocking` to generate blocking APIs callable by Java and Kotlin.
+    rpcCallStyle = 'suspending'
+    // Server only
+    // True for emitted services to implement one interface per RPC.
```
Comment (Egorand) [inline docs/wire_grpc.md]:
```suggestion
    // True for emitted services to generate one interface per RPC. 
```

File: docs/wire_grpc.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    rpcCallStyle = 'suspending'
+    // Server only
+    // True for emitted services to implement one interface per RPC. 
+    singleMethodServices = false
+  }
+}
+```
+
+The generated code varies depending on your RPC role, namely client or server.  
+We'll use the following schema to understand how they differ:
```
Comment (Egorand) [inline docs/wire_grpc.md]:
```suggestion
We'll use the following schema to demonstrate how they differ:
```

File: docs/wire_grpc.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  suspend fun GetFeature(request: Point): Feature
+
+  fun ListFeatures(request: Rectangle, response: SendChannel<Feature>)
+
+  fun RecordRoute(request: ReceiveChannel<Point>): RouteSummary
+
+  fun RouteChat(request: ReceiveChannel<RouteNote>, response: SendChannel<RouteNote>)
+}
+```
+The server can use Kotlin coroutines suspend and channels mechanism to execute suspending network
```
Comment (Egorand) [inline docs/wire_grpc.md]:
```suggestion
The server can use Kotlin coroutines' suspend and Channels mechanisms to execute suspending network
```

File: docs/wire_grpc.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+calls.
+
+### Sample
+
+Check out our [sample project][sampleProject] demonstrating duplex communications between one
```
Comment (Egorand) [inline docs/wire_grpc.md]:
```suggestion
Check out our [sample project][sampleProject] demonstrating bidirectional communication between an
```

File: docs/wire_grpc.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+Android and one server application.
```
Comment (Egorand) [inline docs/wire_grpc.md]:
```suggestion
Android application and a server.
```

**Pull request merged**